src/Elmish.WPF/ViewModel.fs

```diff
@@ -336,6 +344,12 @@ and [<AllowNullLiteral>] internal ViewModel<'model, 'msg>
                ViewModel(m, (fun msg -> toMsg (getId m, msg) |> dispatch), getBindings (), config, chain)
           )
           |> ObservableCollection
+        let modlesById = Dictionary<_,_>(vms.Count)
+        for model in vms |> Seq.map (fun vm -> vm.CurrentModel) do
+          let id = getId model
+          if modlesById.ContainsKey id
+          then failwithf "The getId function is suppose to return distinct IDs for each sub-model, but it returned the same ID %A for the two sub-models %A and %A" id (modlesById.[id]) model
```

Comment:
This should be 'supposed' not 'suppose'. Missing a 'd' :)

Also if the ids are the same, then printing those two ids will not be all that useful. Perhaps a better message would be:

"The getId function should return a unique id for all sub-models, but has returned the id '%A' for more than one item"

Reply:
> This should be 'supposed' not 'suppose'. Missing a 'd' :)

Thanks!  I force pushed the correction.

> Also if the ids are the same, then printing those two ids will not be all that useful.

That is not what is printed.  It prints three things each once:
1. an ID returned from `getId`,
2. an input to `getId` that returns this ID, and
3. a different input to `getId` that returns this same ID.

Reply:
Ah yep I missed the intention - I was just thinking from my use case, where I used the `id` function (as does the samples); will this not return the same text three times?